### PR TITLE
58715_init_model_and_relationship

### PR DESCRIPTION
**Pull request by ahammad-reyad**
Created Models and Relationships

### CODE REVIEW

File: app/models/category.rb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+class Category < ActiveRecord::Base
```
Comment (MuhammadTamzid) [inline app/models/category.rb]: missing relationship

File: app/models/answer.rb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+class Answer < ActiveRecord::Base
+  belongs_to :word
```
Comment (MuhammadTamzid) [inline app/models/answer.rb]: missing relationship

File: app/models/lesson.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+class Lesson < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :category
```
Comment (MuhammadTamzid) [inline app/models/lesson.rb]: missing relationship

File: app/models/word.rb [hunk 4]
```diff
@@ -0,0 +1,3 @@
+class Word < ActiveRecord::Base
+  belongs_to :category
```
Comment (MuhammadTamzid) [inline app/models/word.rb]: missing relationship

File: app/models/answer.rb (hunk 2 above)
Comment (ahammad-reyad) [inline app/models/answer.rb]: added has_many relation

File: app/models/category.rb (hunk 1 above)
Comment (ahammad-reyad) [inline app/models/category.rb]: added missing relationships

File: app/models/lesson.rb (hunk 3 above)
Comment (ahammad-reyad) [inline app/models/lesson.rb]: added missing relationship

File: app/models/word.rb (hunk 4 above)
Comment (ahammad-reyad) [inline app/models/word.rb]: added missing relationships

File: app/models/relationship.rb [hunk 5]
```diff
@@ -0,0 +1,6 @@
+class Relationship < ActiveRecord::Base
+  belongs_to :follower, class_name: "User"
```
Comment (dieunb) [inline app/models/relationship.rb]:
```
+  belongs_to :follower, class_name: User.name
+  belongs_to :followed, class_name: User.name
```

File: app/models/user.rb [hunk 6]
```diff
@@ -0,0 +1,12 @@
+class User < ActiveRecord::Base
+  has_many :activities, dependent: :destroy
+  has_many :lessons, dependent: :destroy
+  has_many :active_relationships, class_name:  "Relationship",
```
Comment (dieunb) [inline app/models/user.rb]:
```
+  has_many :active_relationships, class_name:  Relationship.name,
 +    foreign_key: :follower_id,
 +    dependent: :destroy
 +  has_many :passive_relationships, class_name:  Relationship.name,
 +    foreign_key: :followed_id,
 +    dependent: :destroy
```

**Pull request merged**